pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -613,20 +680,201 @@ func TestDryRunPreemption(t *testing.T) {
+				st.MakePod().Name("p4").UID("p5").Node("node5").Priority(midPriority).Req(largeRes).Obj(),
+			},
+			DisableParallelism: true,
+			expected: [][]Candidate{
+				{
+					&candidate{
+						name: "node1",
+						victims: &extenderv1.Victims{
+							Pods: []*v1.Pod{st.MakePod().Name("p1").UID("p1").Node("node1").Priority(midPriority).Req(largeRes).Obj()},
+						},
+					},
+					&candidate{
+						name: "node2",
+						victims: &extenderv1.Victims{
+							Pods: []*v1.Pod{st.MakePod().Name("p2").UID("p2").Node("node2").Priority(midPriority).Req(largeRes).Obj()},
 						},
-						NumPDBViolations: 1,
 					},
-					name: "node1",
 				},
 			},
-			expectedNumFilterCalled: 4,
+			expectedNumFilterCalled: []int32{4},
+		},
+		{
+			name: "some nodes are not possible candidates, DefaultPreemptionArgs limits to 2",
+			args: &config.DefaultPreemptionArgs{MinCandidateNodesPercentage: 40, MinCandidateNodesAbsolute: 1},
+			registerPlugins: []st.RegisterPluginFunc{
+				st.RegisterPluginAsExtensions(noderesources.FitName, noderesources.NewFit, "Filter", "PreFilter"),
+			},
+			nodeNames: []string{"node1", "node2", "node3", "node4", "node5"},
+			testPods: []*v1.Pod{
+				st.MakePod().Name("p").UID("p").Priority(highPriority).Req(largeRes).Obj(),
+			},
+			initPods: []*v1.Pod{
+				st.MakePod().Name("p1").UID("p1").Node("node1").Priority(veryHighPriority).Req(largeRes).Obj(),
+				st.MakePod().Name("p2").UID("p2").Node("node2").Priority(midPriority).Req(largeRes).Obj(),
+				st.MakePod().Name("p3").UID("p3").Node("node3").Priority(veryHighPriority).Req(largeRes).Obj(),
+				st.MakePod().Name("p4").UID("p4").Node("node4").Priority(midPriority).Req(largeRes).Obj(),
+				st.MakePod().Name("p4").UID("p5").Node("node5").Priority(midPriority).Req(largeRes).Obj(),
+			},
+			DisableParallelism: true,
+			expected: [][]Candidate{
+				{
+					&candidate{
+						name: "node2",
+						victims: &extenderv1.Victims{
+							Pods: []*v1.Pod{st.MakePod().Name("p2").UID("p2").Node("node2").Priority(midPriority).Req(largeRes).Obj()},
+						},
+					},
+					&candidate{
+						name: "node4",
+						victims: &extenderv1.Victims{
+							Pods: []*v1.Pod{st.MakePod().Name("p4").UID("p4").Node("node4").Priority(midPriority).Req(largeRes).Obj()},
+						},
+					},
+				},
+			},
+			expectedNumFilterCalled: []int32{4},
+		},
+		{
+			name: "preemption offset across multiple scheduling cycles",
+			args: &config.DefaultPreemptionArgs{MinCandidateNodesPercentage: 40, MinCandidateNodesAbsolute: 1},
+			registerPlugins: []st.RegisterPluginFunc{
+				st.RegisterPluginAsExtensions(noderesources.FitName, noderesources.NewFit, "Filter", "PreFilter"),
+			},
+			nodeNames: []string{"node1", "node2", "node3", "node4", "node5"},
+			testPods: []*v1.Pod{
+				st.MakePod().Name("tp1").UID("tp1").Priority(highPriority).Req(largeRes).Obj(),
+				st.MakePod().Name("tp2").UID("tp2").Priority(highPriority).Req(largeRes).Obj(),
```

Comment:
Can you add a testPod "tp3" to verify the with slice boundaries? tp3 is expected to come up with candidates [node5, node1], right?

Reply:
Great point! Done.